src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -832,14 +819,63 @@ void WindowImplX11::processEvents()
 
     // Pick out the events that are interesting for this window
     while (XCheckIfEvent(m_display, &event, &checkEvent, reinterpret_cast<XPointer>(m_window)))
-        m_events.push_back(event);
-
-    // Handle the events for this window that we just picked out
-    while (!m_events.empty())
     {
-        event = m_events.front();
-        m_events.pop_front();
-        processEvent(event);
+        // This function implements a workaround to properly discard
+        // repeated key events when necessary. The problem is that the
+        // system's key events policy doesn't match SFML's one: X server will generate
+        // both repeated KeyPress and KeyRelease events when maintaining a key down, while
+        // SFML only wants repeated KeyPress events. Thus, we have to:
+        // - Discard duplicated KeyRelease events when m_keyRepeat is true
+        // - Discard both duplicated KeyPress and KeyRelease events when m_keyRepeat is false
+
+        bool processThisEvent = true;
+
+        // Detect repeated key events
+        while (event.type == KeyRelease)
+        {
+            XEvent nextEvent;
+            if (XCheckIfEvent(m_display, &nextEvent, checkEvent, reinterpret_cast<XPointer>(m_window)))
+            {
+                if (nextEvent.type == KeyPress &&
+                    nextEvent.xkey.keycode == event.xkey.keycode &&
+                    event.xkey.time <= nextEvent.xkey.time &&
+                    nextEvent.xkey.time <= event.xkey.time + 1)
```

Comment:
Can you not assume the same time?

Reply:
Maybe. I just used the same condition as before which was `(event.xkey.time - time < 2)`, but rewrote it to be more explicit, i.e. not rely on unsigned integer wrapping. Involved variables are of type `Time` which is an Xlib typedef to `unsigned int`.